src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -664,20 +664,31 @@ private[internal] trait TypeMaps {
   }
 
   /** A base class to compute all substitutions */
-  abstract class SubstMap[T](from: List[Symbol], to: List[T]) extends TypeMap {
-    // OPT this check was 2-3% of some profiles, demoted to -Xdev
-    if (isDeveloper) assert(sameLength(from, to), "Unsound substitution from "+ from +" to "+ to)
+  abstract class AbstractSubstMap[T >: Null] extends TypeMap {
+    protected def from: List[Symbol] = Nil
+    protected def to: List[T] = Nil
```

Comment:
_Note_: We turned these into `def` because one subclass will back them up on an immutable `val` field, and another classes will back them up with a mutable `var`.